### ISSUE TITLE
Add VPC Endpoint option to storage gateway

### ISSUE THREAD

**Issue by jecafarelli** [terraform-providers/terraform-provider-aws#9920]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Looking for an option to be added to the creation of the storage gateway for a gateway created in a private subnet (using a VPC endpoint)

### New or Affected Resource(s)

* aws_storagegateway_gateway

### Potential Terraform Configuration

```hcl
resource "aws_storagegateway_gateway" "example" {
  gateway_ip_address = "192.168.0.1"
  gateway_name       = "test"
  gateway_timezone   = "GMT"
  gateway_type       = "FILE_S3"
  gateway_vpc_endpoint = "vpce-id.storagegateway.us-east-1.vpce.amazonaws.com"
}
```

### References

https://docs.aws.amazon.com/storagegateway/latest/userguide/create-gateway-file.html#GettingStarted-service-endpoint-file

Code should go before this Line
https://github.com/terraform-providers/terraform-provider-aws/blob/master/aws/resource_aws_storagegateway_gateway.go#L149

And should add `&vpcEndpoint=VPCENDPOINT_VARIABLE` to http request

example:
```http://192.168.0.1/?activationRegion=us-east-1&vpcEndpoint=vpc-id.storagegateway.us-east-1.vpce.amazonaws.com```

Also looks like you need to set it using this method:

https://github.com/aws/aws-sdk-go/blob/master/service/storagegateway/api.go#L11176

**Comment by jecafarelli** [comment]
I created a PR for this code change, and I built and tested it following the guidelines in the contribution page. It is working for me with the manual testing I have done, I just dont have a good way to run the acceptance tests against an aws account so I cant finish.

Creating a valid test to test the VPC endpoint would also be a bit different then the current setup in the test as you would need to create an ec2 instance within a private subnet, create a vpc endpoint, and also have a peering connection from your local network to the private network in aws so you could also hit the ec2 instance with http.

**Comment by tlindsay42** [comment]
Tested this last month successfully. Adding testing details in the PR.

**Comment by hariepuri5** [comment]
Any update on storage gateway to accept a vpc endpoint release??

**Comment by bouazizkhaled** [comment]
Hello any news on vpc endpoint release ?

**Comment by akramWewe** [comment]
Hello,  yes please any news for the release which add this storage gateway option ?

**Comment by bflad** [comment]
Support for the `gateway_vpc_endpoint` argument on the `aws_storagegateway_gateway` resource has been merged and will release with version 2.60.0 of the Terraform AWS Provider, later this week. Thanks to @jecafarelli for the implementation. 👍
